### PR TITLE
Split tests for transforms v2 and prototype

### PR DESCRIPTION
**Pull request by pmeier**
With the branch cut coming around the `torchvision/prototype` namespace will be removed for the release branch. Right now all the tests for the migrated `torchvision.datapoints` and `torchvision.transforms.v2` unconditionally import from `torchvision.prototype` and thus will fail. This PR does the following:

- Move all v2 functionalities from `prototype_common_utils` into `common_utils`. The only things remain in `prototype_common_utils` is the label and one hot label data generation
- Create `test_datapoints` and move everything non-label related there from `test_prototype_datapoints`
- Rename `prototype_transforms_{kernel,dispatcher}_infos` and `test_prototype_transforms_{functional,utils}` into `*transforms_v2*`
- create a new `test_transforms_v2` and move most of `test_prototype_transforms` there. Some things like batch or video transforms stay in `test_prototype_transforms`
- Rename `test_prototype_transforms_consistency` to `test_transforms_v2_consistency` and remove all label related stuff. Furthermore, move test for `FixedSizedCrop` to `test_prototype_transforms`.

cc @vfdev-5 @bjuncek

### CODE REVIEW

File: .github/workflows/test-linux-cpu.yml [hunk 1]
```diff
@@ -54,4 +54,6 @@ jobs:
 
         # Run Tests
         python3 -m torch.utils.collect_env
+        rm -rf torchvision/prototype
+        rm test/*prototype*
```
Comment (pmeier) [inline .github/workflows/test-linux-cpu.yml:58]: This comes from 67fdc6d1b9ecd9bf955af86da60fa749bb05fb7f that should be reverted. With this we can check if I didn't screw up anything and branch cut can happen smoothly

File: test/datasets_utils.py [hunk 2]
```diff
@@ -170,23 +170,6 @@ def wrapper(self):
     return wrapper
 
 
-def combinations_grid(**kwargs):
```
Comment (pmeier) [inline test/datasets_utils.py:173]: I've moved this to `common_utils` since it is no longer just used by the dataset tests.

File: test/test_transforms_v2_consistency.py [hunk 3]
```diff
@@ -1102,11 +1103,12 @@ def make_datapoints(self, with_mask=True):
...
-                det_transforms.FixedSizeCrop((1024, 1024), fill=0),
-                prototype_transforms.FixedSizeCrop((1024, 1024), fill=0),
-                {},
-            ),
+            # FIXME
+            # (
+            #     det_transforms.FixedSizeCrop((1024, 1024), fill=0),
+            #     prototype_transforms.FixedSizeCrop((1024, 1024), fill=0),
+            #     {},
+            # ),
```
Comment (pmeier) [inline test/test_transforms_v2_consistency.py:1111]: Oops. This is handled, but I forgot to remove the comment.

```suggestion
```

**Comment by pmeier** [comment]
> Why move transforms-related tests into `test_datapoints`? Is there a typo?

Yes. It should say

> Create `test_datapoints` and move everything non-label related there from `test_prototype_datapoints`

Will update my top comment.

**Comment by pmeier** [comment]
Except for Python 3.8, all unittest jobs on macOS are failing due to insufficient resources on the runner: https://app.circleci.com/pipelines/github/pytorch/vision/23382/workflows/05e31fc3-0271-4ad4-8d48-a0b82b3e099f/jobs/1834265?invite=true#step-109-223. 

We are using 

https://github.com/pytorch/vision/blob/dfa81ce4191ea8dc96a5b509ed27c6938a8b853d/.circleci/config.yml.in#L904

which [translates to](https://circleci.com/docs/configuration-reference/#macos-execution-environment)

> 8 @ 2.7 GHz | 16GB

There is also `macos.x86.metal.gen1`

> 12 @ 3.2 GHz | 32GB

Will try this one.

**Comment by pmeier** [comment]
Unittest on Windows and Linux are green though, so I'' revert 67fdc6d1b9ecd9bf955af86da60fa749bb05fb7f.

**Comment by pmeier** [comment]
It seems we don't have access to the larger runner: https://app.circleci.com/pipelines/github/pytorch/vision/23385/workflows/979cac8c-f706-44cc-8a7b-8102a58ebbe4/jobs/1834346

> Job was rejected because resource class macos.x86.metal.gen1, image xcode:14.0 is not a valid resource class
